migrations/statictypes/statictype_migration.go

```diff
@@ -177,20 +177,17 @@ func (m *StaticTypeMigration) maybeConvertStaticType(
 	// Parse of the migration, e.g. the intersection type migration depends on the parent type.
 	// For example, `{Ts}` in `&{Ts}` is migrated differently from `{Ts}`.
 
-	if parentType == nil {
+	migratedTypeCache := m.migratedTypeCache
+	staticTypeID := staticType.ID()
 
```

Comment:
is this not needed now? 🤔

Reply:
I've added it back in [21f06f7](https://github.com/onflow/cadence/pull/3329/commits/21f06f73fc3d5364d49f0a08064cd79015045780). I can't come up with an example where it is needed, but neither can I convince myself there isn't one. Let's conservatively do it only for the root type, where we can always be sure it is correct. As a side-effect, it might also reduce traffic on and the size of the cache.